### ISSUE TITLE
Long running (remote) ansible provisioner hangs: dial tcp 127.0.0.1:<port>: connect: connection refused

### ISSUE THREAD

**Issue by timblaktu** [hashicorp/packer#10700]
This is a follow up to [#6877](https://github.com/hashicorp/packer/issues/6877#) which appears to be the same problem, as my long-running ansible task is simply doing a huge apt install on a Debian guest. Also I am NOT specifying any provisioner options other than user, playbook_file, galaxy_file, and extra_arguments. That issue was closed without a resolution that I could understand. Someone posted a workaround of using the ansible-local provisioner, but since I'm not able to comment on that closed issue, I wanted to ask the community:

**What are your options for fixing the problem of a long-running ansible provisioner that it hanging?**

There was a linked discussion started by @SwampDragons about [removing the "proxy" in the ansible provisioner](https://github.com/hashicorp/packer/pull/8625), which was done, tested and merged a year ago, but I'm not understanding what the meaning of this is, WRT the problem I'm having.

**Comment by SwampDragons** [comment]
There are a lot of linked issues on that PR, but the one that gives the best context is this one: https://github.com/hashicorp/packer/issues/7667

The tl;dr is that Packer by default creates a local proxy server so that Packer's SSH communicator can be used to send ansible commands (which would normally be called locally and create their own ssh connection). This proxy allows Packer to use ansible against things it's provisioning that don't have their own IP, like docker containers. 

But it's not a perfect solution. Particularly, when ansible 2.8 was released, it started doing this ssh pipelineing thing that made the proxy hang for certain users. So we added an option to the ansible provisioner to choose _not_ to use the proxy or communicator, and instead just shell out to ansible directly using a custom-built inventory file.

**Comment by SwampDragons** [comment]
The relevant option is documented here: https://www.packer.io/docs/provisioners/ansible#use_proxy

Basically, when you see hangs with the Ansible provisioner, it's worth trying to turn off the proxy (set `use_proxy = false`) to see if that un-sticks you.

**Comment by milgelren** [comment]
Hi @timblaktu, just following up on this issue. Since we haven’t heard back in a while and there hasn’t been any recent activity, we’re going to go ahead and close this for now . If you’re still experiencing this problem or have any updates, feel free to comment or reopen the issue—happy to take another look. Thanks for your contribution!
